bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/client/WebSocketConnectionImpl.java

```diff
@@ -0,0 +1,175 @@
+        this.pingPeriod = pingPeriod;
+        this.pingHandle = executor.scheduleWithFixedDelay(this::sendPing, pingPeriod.dividedBy(2).toMillis(),
+                pingPeriod.toMillis(), TimeUnit.MILLISECONDS);
+        this.watchDogHandle = executor.scheduleWithFixedDelay(this::checkConnection, pingPeriod.toMillis(),
+                pingPeriod.toMillis(), TimeUnit.MILLISECONDS);
+    }
+
+    @Override
+    public void close() {
+        var session = sessionRef.getAndSet(null);
+        if (session != null) {
+            session.close();
+        }
+        watchDogHandle.cancel(true);
+        pingHandle.cancel(true);
+    }
+
+    @Override
+    public void observeProperty(@NotNull String propertyName, @NotNull BiConsumer<String, Object> listener) {
+        propertyChangedListeners.put(propertyName, listener);
+    }
+
+    @Override
+    public void onWebSocketConnect(Session session) {
+        sessionRef.set(session); // save websocket session to be able to send ping
+    }
+
+    @Override
+    public void onWebSocketPing(ByteBuffer payload) {
+    }
+
+    @Override
+    public void onWebSocketPong(ByteBuffer payload) {
+        lastTimeReceived.set(Instant.now());
+    }
+
+    @Override
+    public void onWebSocketBinary(byte[] payload, int offset, int len) {
+    }
+
+    @Override
+    public void onWebSocketText(String message) {
+        try {
+            var propertyStatus = gson.fromJson(message, PropertyStatusMessage.class);
+            if ((propertyStatus != null) && (propertyStatus.messageType != null)
+                    && (propertyStatus.messageType.equals("propertyStatus"))) {
+                for (var propertyEntry : propertyStatus.data.entrySet()) {
+                    var listener = propertyChangedListeners.getOrDefault(propertyEntry.getKey(),
+                            EMPTY_PROPERTY_CHANGED_LISTENER);
+                    try {
+                        listener.accept(propertyEntry.getKey(), propertyEntry.getValue());
+                    } catch (RuntimeException re) {
+                        logger.warn("calling property change listener {} failed. {}", listener, re.getMessage());
+                    }
+                }
+            } else {
+                logger.debug("Ignoring received message of unknown type: {}", message);
+            }
+        } catch (JsonSyntaxException se) {
+            logger.warn("received invalid message: {}", message);
+        }
+    }
+
+    @Override
+    public void onWebSocketClose(int statusCode, String reason) {
+        onWebSocketError(new IOException("websocket closed by peer. " + Optional.ofNullable(reason).orElse("")));
+    }
+
+    @Override
+    public void onWebSocketError(Throwable cause) {
+        onError(cause.getMessage());
+    }
+
+    private void onError(String message) {
+        try {
+            errorHandler.accept(message);
+        } catch (Exception e) {
+            logger.warn("error occurred by performing on disconnect", e);
```

Comment:
Do you log the stack trace by intention?

Reply:
handling exception has been removed